segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -187,5 +239,8 @@ public void close() {
         // Shut down all event loops to terminate all threads.
         bossGroup.shutdownGracefully();
         workerGroup.shutdownGracefully();
+        if (!executor.isShutdown()) {
```

Comment:
This is the wrong way to shut down an executor. See `ExecutorServiceHelpers.shutdown`

Reply:
I've modified the code to avoid using the executor, as per another suggestion below.